src/Core/Vault/Services/Implementations/CipherService.cs

```diff
@@ -999,18 +1000,53 @@ private async Task ValidateCipherCanBeShared(
             throw new BadRequestException("Could not find organization.");
         }
 
-        if (hasAttachments && !org.MaxStorageGb.HasValue)
+        var migrateFeatureEnabled = _featureService.IsEnabled(FeatureFlagKeys.MigrateMyVaultToMyItems);
+
+        // Ignore storage limits if the organization has data ownership policy enabled.
+        // Allows users to seamlessly migrate their data into the organization without being blocked by storage limits.
+        // Organization admins will need to manage storage after migration should overages occur.
+        var ignoreStorageLimits = migrateFeatureEnabled && await OrganizationDataOwnershipPolicyEnabledAsync(sharingUserId, org);
+
+        if (!ignoreStorageLimits)
+        {
+            if (hasAttachments && !org.MaxStorageGb.HasValue)
+            {
+                throw new BadRequestException("This organization cannot use attachments.");
+            }
+
+            var storageAdjustment = attachments?.Sum(a => a.Value.Size) ?? 0;
+            if (org.StorageBytesRemaining() < storageAdjustment)
+            {
+                throw new BadRequestException("Not enough storage available for this organization.");
+            }
+        }
+
+        ValidateCipherLastKnownRevisionDate(cipher, lastKnownRevisionDate);
+    }
+
+    /// <summary>
+    /// Checks if the Organization Data Ownership Policy is enabled for the given user and organization.
+    /// </summary>
+    private async Task<bool> OrganizationDataOwnershipPolicyEnabledAsync(Guid userId, Organization organization)
+    {
+        if (!organization.UsePolicies)
         {
-            throw new BadRequestException("This organization cannot use attachments.");
+            return false;
         }
 
-        var storageAdjustment = attachments?.Sum(a => a.Value.Size) ?? 0;
-        if (org.StorageBytesRemaining() < storageAdjustment)
+        if (_featureService.IsEnabled(FeatureFlagKeys.PolicyRequirements))
         {
-            throw new BadRequestException("Not enough storage available for this organization.");
+            var requirement = await _policyRequirementQuery.GetAsync<OrganizationDataOwnershipPolicyRequirement>(userId);
+
+            return requirement.State == OrganizationDataOwnershipState.Enabled &&
+                   requirement.EnforcedByOrg(organization.Id);
         }
 
-        ValidateCipherLastKnownRevisionDate(cipher, lastKnownRevisionDate);
+        var policies =
+            await _policyService.GetPoliciesApplicableToUserAsync(userId,
+                PolicyType.OrganizationDataOwnership);
+
+        return policies.Any(p => p.OrganizationId == organization.Id);
     }
```

Comment:
This can be simplified:

- put it behind your migration flag, not the PolicyRequirements flag (that is for migrating existing code, but this is new code, so it can go behind the flag for the feature it relates to)
- therefore, you can also drop the `policyService` call, just use the new PolicyRequirementQuery
- no need to check the requirement.State, just use the new method you added